pkg/client/client.go

```diff
@@ -277,28 +277,47 @@ func New(ctx context.Context, options ...Option) (*Client, error) {
 	if c.DSN == "" {
 		c.Logger.Warn("missing DSN, some commands won't work")
 	} else {
-		c.pool, err = CreateDatabase(ctx, c.DSN)
+		c.db, err = sdkdb.New(ctx, c.Logger, c.DSN)
 		if err != nil {
 			return nil, err
 		}
-		if err := ValidatePostgresVersion(ctx, c.pool, MinPostgresVersion); err != nil {
-			c.Logger.Warn("postgres validation warning", "err", err)
+
+		var dt schema.DialectType
+		dt, c.dialectExecutor, err = database.GetExecutor(c.Logger, c.DSN, c.HistoryCfg)
+		if err != nil {
+			return nil, fmt.Errorf("getExecutor: %w", err)
 		}
-	}
-	// migrate cloudquery core tables to latest version
-	if c.DSN != "" {
```

Comment:
maybe time to move this to a helper function? the New has become quite big.

Reply:
yes it has, fixed